TombEngine/Sound/sound.cpp

```diff
@@ -1056,26 +1070,27 @@ int GetShatterSound(int shatterID)
 		return SFX_TR4_SMASH_ROCK;
 }
 
+// TODO: Revise flags.
 void PlaySoundSources()
 {
 	static constexpr int PLAY_ALWAYS    = 0x8000;
 	static constexpr int PLAY_BASE_ROOM = 0x4000;
 	static constexpr int PLAY_FLIP_ROOM = 0x2000;
 
-	for (size_t i = 0; i < g_Level.SoundSources.size(); i++)
+	for (const auto& soundSource : g_Level.SoundSources)
 	{
-		const auto& sound = g_Level.SoundSources[i];
-
-		int group = sound.Flags & 0x1FFF;
+		int group = soundSource.Flags & 0x1FFF;
 
 		if (group >= MAX_FLIPMAP)
 			continue;
 
-		if (!FlipStats[group] && (sound.Flags & PLAY_FLIP_ROOM))
-			continue;
-		else if (FlipStats[group] && (sound.Flags & PLAY_BASE_ROOM))
+		bool flipStat = FlipStats[group];
```

Comment:
I am not sure why you need a temp variable if array is accessed only twice and is already readable (it's not like we have an array deeply hidden in some class like `GameEngine.FlipmapController->FilpStats[flipInfo.group]`).

Reply:
Sure!